content/iOS/Practices/iOS-practices.md

```diff
@@ -0,0 +1,108 @@
+
+It is also a nice practice to separate private functions into a private extension, so that a client at a glance at the class can differentiate the interface from the implementation.  
+
+## Resource Management 
+
+As we mention before we want to have a folder in the root of the project for the resources our app will need, there are many types of resources so its helpful to organize them in different folders according to their type, for example, you may have: property lists, images, gifs, JSONs... ant the list goes on.
+
+For images, Xcode provides a very helpful resource bundle named xassets. 
+Here we'll store the images our app uses, if your app has a fancy UI and uses loads of them is good to have them grouped in folders that describe their use. 
+
+Images may be heavy, and we want our apps to be as small in size as possible, so if we find yourself having to add the same image in different colors its recommended to use them as templates and color them in code instead of adding different versions of the same image.  
+
+Also, the naming of the resource itself should be descriptive enough so that you'll find it quickly and makes sense in the place you reference it. 
+
+## Storyboards and xibs vs Code
+
+Storyboards and xibs are great when you start learning to code in swift. 
+The visual help does a lot kickstarting your designs and helping you understand `NSLayoutConstraints`.
+Also if well handled, the way they represent the flow helps you understand a lot of the app by having a glance at it.
+
+However... They are heavy and often take a lot to load, connections between storyboard views and code are not checked in compiler time so you may have crashes if not careful, when working in a team the merge conflicts will be very hard to solve, its very annoying to review the autogenerated code in Pull Requests, it doesn't favor reuse and componentization, and most importantly there's not a "single source of truth" for your UI code. UI and navigation code will be divided between storyboards and your view controllers making it way harder to understand and maintain. 
+
+So my inclination here is to do all UI work in code.
+Once you understand how `NSLayoutConstraint` works, identify the code you are repeating, and develop helpers to avoid that, you'll find yourself accomplishing results faster, and with fewer lines of code. 
+Code review processes will go much smoother, you'll find much more opportunities to reuse, and you'll find all UI work in a single place and avoid jumping from storyboards or xibs to code every minute.
+
+If you loved the way you were able to quickly grasp your app's flow in the storyboard my recommendation would be to include flow diagrams in the code base, which may save precious time in the future!  
```

Comment:
I am not sure if recommending this. I personally haven't done this myself and I think this would add the extra work of updating those on every flow modification. But I don't really know really.

Reply:
Haven't done this myself lol. But if someone's concern is to lose the visual sense of the app flow when removing storyboards this may be a good suggestion

Reply:
Cool, in that case I would clarify that that way may require extra work as well. If they choose to do so they need to know the implications :)